src/main/config/config_eeprom.c

```diff
@@ -168,6 +170,70 @@ static uint32_t getFLASHSectorForEEPROM(void)
 }
 #endif
 
+#if defined(STM32F7)
+
+// FIXME: HAL for now this will only work for F4/F7 as flash layout is different
+void writeEEPROM(void)
+{
+    // Generate compile time error if the config does not fit in the reserved area of flash.
+    BUILD_BUG_ON(sizeof(master_t) > ((uint32_t)&__config_end - (uint32_t)&__config_start));
+
+    HAL_StatusTypeDef status;
+    uint32_t wordOffset;
+    int8_t attemptsRemaining = 3;
+
+    suspendRxSignal();
+
+    // prepare checksum/version constants
+    masterConfig.version = EEPROM_CONF_VERSION;
+    masterConfig.size = sizeof(master_t);
+    masterConfig.magic_be = 0xBE;
+    masterConfig.magic_ef = 0xEF;
+    masterConfig.chk = 0; // erase checksum before recalculating
+    masterConfig.chk = calculateChecksum((const uint8_t *) &masterConfig, sizeof(master_t));
+
+    // write it
+    /* Unlock the Flash to enable the flash control register access *************/
+    HAL_FLASH_Unlock();
+    while (attemptsRemaining--)
+    {
+        /* Fill EraseInit structure*/
```

Comment:
This has to be changed to respect the linker script and put configuration into FLASH_CONFIG